File: web-fit-vue/src/api/meetingBooking.ts
import request from '@/utils/request'
import type { ApiResult } from './loginRecord'

export interface MeetingBooking {
  id: string
  roomId: string
  bookingDate: string
  startSlot: number
  endSlot: number
  empNo: string
  empName: string
  meetingTitle: string
  attendees: string
  isWeeklyFix: number
  groupId: string
  status: number
  createTime: string
  updateTime: string
}

/** 看板响应 */
export interface BoardResponse {
  rooms: RoomBoard[]
}

export interface RoomBoard {
  roomId: string
  roomName: string
  capacity: number
  facilities: string[]
  photoUrl: string
  slots: SlotInfo[]
}

export interface SlotInfo {
  slot: number
  timeLabel: string
  type: 'FREE' | 'MY_BOOKING' | 'BOOKED' | 'ADMIN_LOCK'
  booking: BookingInfo | null
  lock: LockInfo | null
}

export interface BookingInfo {
  bookingId: string
  empNo: string
  empName: string
  meetingTitle: string
  attendees: string
  isWeeklyFix: boolean
  groupId: string
}

export interface LockInfo {
  lockId: string
  reason: string
  deptName: string
}

/** 创建预定 */
export function createBooking(params: {
  roomId: string
  bookingDate: string
  startSlot: number
  endSlot: number
  empName?: string
  meetingTitle?: string
  attendees?: string
  weeklyWeeks?: number
}): Promise<ApiResult<MeetingBooking[]>> {
  return request.post('/meeting-booking', params)
}

/** 修改预定信息 */
export function updateBookingInfo(id: string, data: {
  meetingTitle?: string
  attendees?: string
}): Promise<ApiResult<MeetingBooking>> {
  return request.put(`/meeting-booking/${id}`, data)
}

/** 取消单次预定 */
export function cancelBooking(id: string): Promise<ApiResult<null>> {
  return request.delete(`/meeting-booking/${id}`)
}

/** 一键取消后续所有周期约 */
export function cancelGroupFuture(groupId: string, fromDate: string): Promise<ApiResult<number>> {
  return request.delete(`/meeting-booking/group/${groupId}`, { params: { fromDate } })
}

/** 查询指定会议室某日预定 */
export function getRoomBookings(roomId: string, date: string): Promise<ApiResult<MeetingBooking[]>> {
  return request.get(`/meeting-booking/room/${roomId}`, { params: { date } })
}

/** 全局看板 */
export function getBoard(date: string): Promise<ApiResult<BoardResponse>> {
  return request.get('/meeting-booking/board', { params: { date } })
}

File: java-fit-server/src/main/java/com/fit/entity/MeetingBooking.java
package com.fit.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("meeting_booking")
public class MeetingBooking {

    @TableId(type = IdType.ASSIGN_ID)
    private String id;

    /** 关联会议室ID */
    private String roomId;

    /** 预定日期 */
    private LocalDate bookingDate;

    /** 起始时段编号（0=08:00, 1=08:30, ...） */
    private Integer startSlot;

    /** 结束时段编号（不包含） */
    private Integer endSlot;

    /** 预定人工号 */
    private String empNo;

    /** 预定人姓名（冗余） */
    private String empName;

    /** 会议名称（选填） */
    private String meetingTitle;

    /** 参会人（选填） */
    private String attendees;

    /** 是否固定周期约：0-否，1-是 */
    private Integer isWeeklyFix;

    /** 周期约组ID（同一批生成共享） */
    private String groupId;

    /** 状态：1-正常，0-已取消 */
    private Integer status;

    /** 创建时间 */
    private LocalDateTime createTime;

    /** 更新时间 */
    private LocalDateTime updateTime;
}
